### ISSUE TITLE
comma-dangle: Not accurate when last array item is wrapped in parentheses

### ISSUE THREAD

**Issue by Met48** [eslint/eslint#4195]
Using ESLint v1.7.1, the `comma-dangle` rule is not accurate when the last array item has (unnecessary) parentheses. It will always warn when the rule option is set to `"always"` and never warn when the rule option is set to `"never"`.

Examples:

```
/* eslint comma-dangle: [2, "always"] */
[(1),]
```

Expected: No errors
Actual: `2:4  error  Missing trailing comma  comma-dangle`

```
/* eslint comma-dangle: [2, "never"] */
[(1),]
```

Expected: `2:4  error  Unexpected trailing comma  comma-dangle`
Actual: No errors

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Yeah, that looks like a bug. Probably confused by the parens.

**Comment by nzakas** [comment]
Working on this.
